CommunicationLayer/MessagingFramework/MessageDecodingHelpers.cpp

```diff
@@ -27,23 +27,47 @@
 
 namespace
 {
-   const int NUMBER_OF_BYTES_IN_FLOAT = 4;
+    const int UNION_SIZE = 4;
 
-   union FloatCharTranslator
-   {
-      float floatData;
-      char charData[NUMBER_OF_BYTES_IN_FLOAT];
-   };
+    union DataFromCharTranslator
+    {
+        unsigned short unsignedShortData;
+        float floatData;
+        char charData[MessageDecodingHelpers::numberOfBytesInData(UNION_SIZE)];
+    }
 }
 
-float MessageDecodingHelpers::getFloat(
-   const QByteArray& data, int startIndex)
+float MessageDecodingHelpers::getFloat(const QByteArray& data, int startIndex)
 {
-   FloatCharTranslator dataUnion;
-   for (int i = 0; i < NUMBER_OF_BYTES_IN_FLOAT; i++)
-   {
-      dataUnion.charData[i] = data.at(i + startIndex);
-   }
+   return getData(data, startIndex, Type::FLOAT).floatData;
+}
+
+unsigned short MessageDecodingHelpers::getUnsignedShort(const QByteArray& data, int startIndex)
+{
+    return getData(data, startIndex, TYPE::UNSIGNED_SHORT).unsignedShortData;
+}
 
-   return dataUnion.floatData;
+void MessageDecodingHelpers::getData(const QByteArray& data, int startIndex, Type type)
+{
+    DataFromCharTranslator dataUnion;
+    for (int i = 0; i < numberOfBytesInData(type); i++)
+    {
+        dataUnion.charData[i] = data.at(i = startIndex);
+    }
+    return dataUnion;
+}
+
+const int MessageDecodingHelpers::numberOfBytesInData(const Type& type)
+{
+    switch (type)
+    {
+    case FLOAT:
+        return sizeof(float);
+        break;
+    case UNSIGNED_SHORT:
+        return sizeof(unsigned short);
+        break;
+    default:
+        throw "Unhandled data type!";
+    }
```

Comment:
Because this is onboard critical softwre that we can't really restart easliy in "production", a debug statement would be better I think.  Not much in terms of "standards" but more dependent on the sitaution. If this was on a client application, I would be more inclined to crash the program.